Mathlib/NumberTheory/Cyclotomic/Three.lean

```diff
@@ -0,0 +1,115 @@
+/-
+Copyright (c) 2024 Riccardo Brasca. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Riccardo Brasca, Pietro Monticone
+-/
+
+import Mathlib.NumberTheory.Cyclotomic.Embeddings
+import Mathlib.NumberTheory.Cyclotomic.Rat
+import Mathlib.NumberTheory.NumberField.Units.DirichletTheorem
+
+/-!
+# Third Cyclotomic Field
+We gather various results about the third cyclotomic field. The following notations are used in this
+file: `K` is a number field such that `IsCyclotomicExtension {3} ℚ K`, `ζ` is any primitive `3`-rd
+root of unity in `K`, `η` is the element in the ring of integers corresponding to `ζ` and
+`λ = η - 1`.
+
+## Main results
+* `IsCyclotomicExtension.Rat.Three.Units.mem`: Given a unit `u : (𝓞 K)ˣ`, we have that
+`u ∈ {1, -1, η, -η, η^2, -η^2}`.
+
+* `IsCyclotomicExtension.Rat.Three.eq_one_or_neg_one_of_unit_of_congruent`: Given a unit
+`u : (𝓞 K)ˣ`, if `u` is congruent to an integer modulo `3`, then `u = 1` or `u = -1`.
+
+This is a special case of the so-called *Kummer's lemma* (see for example
+https://kconrad.math.uconn.edu/blurbs/gradnumthy/kummer.pdf).
+-/
+
+open NumberField Units InfinitePlace nonZeroDivisors Polynomial
+
+namespace IsCyclotomicExtension.Rat.Three
+
+variable {K : Type*} [Field K] [NumberField K] [IsCyclotomicExtension {3} ℚ K]
+variable {ζ : K} (hζ : IsPrimitiveRoot ζ ↑(3 : ℕ+)) (u : (𝓞 K)ˣ)
+local notation3 "η" => hζ.toInteger
+local notation3 "λ" => hζ.toInteger - 1
+
+/-- Let `u` be a unit in `(𝓞 K)ˣ`, then `u ∈ {1, -1, η, -η, η^2, -η^2}`. -/
+theorem Units.mem : ↑u ∈ ({1, -1, η, -η, η ^ 2, -η ^ 2} : Set (𝓞 K)) := by
```

Comment:
Using a `List` instead of a `Finset` is further from the informal statement, but may make this more convenient to use.

Reply:
Can you elaborate a bit? `fin_cases` does not work even for a list, and I don't see any reason why lists should be better (actually, there is even one case more to consider in the proof, when `u ∈ []`).